packages/babel-parser/src/parser/statement.js

```diff
@@ -838,14 +847,19 @@ export default class StatementParser extends ExpressionParser {
     directives: ?(N.Directive[]),
     topLevel: boolean,
     end: TokenType,
+    afterBlockParse?: (hasStrictModeDirective: boolean) => void,
   ): void {
+    const octalPositions = [];
     let parsedNonDirective = false;
-    let oldStrict;
-    let octalPosition;
+    let oldStrict = null;
 
     while (!this.eat(end)) {
-      if (!parsedNonDirective && this.state.containsOctal && !octalPosition) {
-        octalPosition = this.state.octalPosition;
+      // Track octal literals that occur before a "use strict" directive.
+      if (!parsedNonDirective && this.state.containsOctal) {
+        // This octal literal will always occur in a directive and will
+        // therefore be escaped. Subtract two from the current position to
+        // account for the "\" character.
+        octalPositions.push(this.state.pos - 2);
```

Comment:
What if it's not at the end of the string, or if it has more than one digit?

`"foo \03 bar"`

Reply:
Maybe we could remove `containsOctal`, since it's the same as `octalPosition !== null`.

Reply:
Goooooood call. I guess another option could be to not track this at all and use a regex check on directives? Not sure how big a hit that would be performance-wise, but it would allow us to remove all this extra "global" state.

Reply:
It would probably be ok, because it super rare to have another directive before `"use strict"` anyway.

Reply:
After exploring this some more, I think we should just go with removing `containsOctal` for now. Removing global state doesn't feel like a good enough reason to recalculate the position, imo.

Reply:
I ended up changing `octalPosition` to `octalPositions` and made it an array, because the current behavior only reports the last octal literal found in a directive (since it doesn't get checked until after the token has been parsed). With an array, we can track the positions of multiple octal literals and warn on each of them.